src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateHisoryClone = [];
+
+  for (const obj of actions) {
```

Comment:
it's better to add more meaningful variable names
```suggestion
  for (const action of actions) {
```